### PR TITLE
Added version diff section to generate-docs

### PR DESCRIPTION
**Pull request by yaakovpraisler**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/37629

## Description
Added version differences section to readme in generate-docs.

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 * Fixed **format** to use git as a default value.
 * Updated the **upload** command to support reports.
 * Fixed an issue where **generate-docs** command was displaying 'None' when credentials parameter display field configured was not configured.
+* Added a version differences section to readme in **generate-docs** command and added the `--docs-format` flag in **integration-diff** command to get the output in readme format.
```
Comment (roysagi) [inline CHANGELOG.md:13]: separate to 2 notes.

File: demisto_sdk/commands/integration_diff/integration_diff_detector.py [hunk 2]
```diff
@@ -243,22 +308,151 @@ def check_if_element_exist(element_to_check, list_of_elements, field_to_check) -
...
         Prints the missing elements report.
+        """
+
+        for missing_type in self.missing_items_report:
+            click.secho(f"\nMissing {missing_type}:\n", fg='bright_red')
+
+            for item in self.missing_items_report[missing_type]:
+                click.secho(item['message'] + "\n", fg='bright_red')
+
+    def print_items_in_docs_format(self):
```
Comment (roysagi) [inline demisto_sdk/commands/integration_diff/integration_diff_detector.py:322]: this method is way too long please split it to more maintainable sub-methods in accordance with the sections of the file.

File: demisto_sdk/commands/integration_diff/integration_diff_detector.py [hunk 3]
```diff
@@ -243,22 +308,151 @@ def check_if_element_exist(element_to_check, list_of_elements, field_to_check) -
...
+        """
+        Prints the version differences report in docs format so the user will can copy it to the README file.
+        """
+
+        old_yaml_data = get_yaml(self.old)
+        new_yaml_data = get_yaml(self.new)
```
Comment (roysagi) [inline demisto_sdk/commands/integration_diff/integration_diff_detector.py:328]: you can save the 2 file data when initing that way there is no need to reload them again.

Comment (yaakovpraisler) [reply demisto_sdk/commands/integration_diff/integration_diff_detector.py:328]: done.

File: demisto_sdk/commands/generate_docs/generate_integration_doc.py [hunk 4]
```diff
@@ -310,6 +316,95 @@ def generate_single_command_section(cmd: dict, example_dict: dict, command_permi
...
+    Arguments:
+        version : The integration version.
+
+    Returns:
+        List of the section lines.
+    """
+
+    section = [
+        '',
+        f'~~~~~~~~~~For the full V{version} important information run `demisto-sdk integration-diff -n '
```
Comment (roysagi) [inline demisto_sdk/commands/generate_docs/generate_integration_doc.py:332]:
```suggestion
        f'~~~~~~~~~~To get a sugggestion for the full *V{version} important information* section run: `demisto-sdk integration-diff -n '
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -17,6 +17,8 @@
 * Added to the **update-release-notes** command automatic addition to git when new release notes file is created.
 * Added validation to ensure that integrations, scripts, and playbooks do not contain the entity type in their names.
 * Added the **convert** command to convert entities between XSOAR versions.
+* Added a version differences section to readme in **generate-docs** command.
+* Added the `--docs-format` flag in **integration-diff** command to get the output in readme format.
```
Comment (roysagi) [inline CHANGELOG.md:21]:
```suggestion
* Added the *--docs-format* flag in the **integration-diff** command to get the output in README format.
```

**Review by roysagi — COMMENTED** [review]
@yaakovpraisler - this looks good, will wait for the demo to see final changes.

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,8 @@
 * Added a validation to ensure integrations are not skipped and at least one test playbook is not skipped for each integration or script.
 * Added to the Content Tests support for `context_print_dt`, which queries the incident context and prints the result as a json.
 * Added new validation for the `xsoar_config.json` file in the **validate** command.
+* Added a version differences section to readme in **generate-docs** command.
```
Comment (roysagi) [inline CHANGELOG.md:7]: please note the 2 additional flags as well.

File: demisto_sdk/commands/common/errors.py [hunk 7]
```diff
@@ -1331,6 +1332,11 @@ def readme_equal_description_error():
     def readme_contains_demisto_word(line_nums):
         return f'Found the word \'Demisto\' in the readme content in lines: {line_nums}.'
 
+    @staticmethod
+    @error_code_decorator
+    def template_sentence_in_readme(line_nums):
+        return f"Found a generic sentence in the readme content in lines: {line_nums}."
```
Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:1338]:
```suggestion
        return f"Please update the integration version differences section in lines: {line_nums}."
```

File: demisto_sdk/commands/generate_docs/generate_integration_doc.py [hunk 8]
```diff
@@ -114,7 +118,14 @@ def generate_integration_doc(
         else:
             docs = []  # type: list
             docs.extend(add_lines(yml_data.get('description')))
-            docs.extend(['This integration was integrated and tested with version xx of {}'.format(yml_data['name'])])
+            docs.extend(['This integration was integrated and tested with version xx of {}'.format(yml_data['name']), ''])
+            # Checks if the integration is a new version
+            integration_version = re.findall("[vV][2-9].yml$", input_path)
```
Comment (roysagi) [inline demisto_sdk/commands/generate_docs/generate_integration_doc.py:123]: why not use the is v2 method from tools? if it does not detect things above v2 lets update the method and still use it here.

File: demisto_sdk/commands/integration_diff/integration_diff_detector.py [hunk 9]
```diff
@@ -95,56 +137,112 @@ def check_command_arguments(self, new_command, old_command):
...
+
+        Return:
+            A description of the change.
+        """
+
+        if field in ['defaultValue', 'defaultvalue']:
+            return f'The default value changed to \'{value}\''
+
+        elif field == 'required':
+            return 'Is Now required'
```
Comment (roysagi) [inline demisto_sdk/commands/integration_diff/integration_diff_detector.py:223]:
```suggestion
            return 'Is now required'
```

**Review by roysagi — APPROVED** [review]
Looks good see my minor corrections

File: CHANGELOG.md (hunk 6 above)
Comment (yaakovpraisler) [reply CHANGELOG.md:7]: done.

File: demisto_sdk/commands/common/errors.py (hunk 7 above)
Comment (yaakovpraisler) [reply demisto_sdk/commands/common/errors.py:1338]: done.

File: demisto_sdk/commands/generate_docs/generate_integration_doc.py (hunk 8 above)
Comment (yaakovpraisler) [reply demisto_sdk/commands/generate_docs/generate_integration_doc.py:123]: done.

File: demisto_sdk/commands/integration_diff/integration_diff_detector.py (hunk 9 above)
Comment (yaakovpraisler) [reply demisto_sdk/commands/integration_diff/integration_diff_detector.py:223]: done.

File: CHANGELOG.md [hunk 10]
```diff
@@ -4,6 +4,9 @@
...
+* Added the *--docs-format* flag in the **integration-diff** command to get the output in README format.
+* Added the *--input-old-version* and *--skip-breaking-changes* flags in the **generate-docs** command to get the detailing changes in the section and to skip the breaking changes section.
```
Comment (roysagi) [inline CHANGELOG.md:9]:
```suggestion
* Added the *--input-old-version* and *--skip-breaking-changes* flags in the **generate-docs** command to get the details for the breaking section and to skip the breaking changes section.
```

**Pull request merged**